src/Education/Commands/ShowRuleHelpFromErrorList.cs

```
@@ -0,0 +1,91 @@
+    internal class ShowHelpFromErrorList : IShowHelpFromErrorList
+    {
+        private readonly IThreadHandling threadHandling;
+        private readonly IEducation education;
+        private readonly IErrorListHelper errorListHelper;
+
+        private readonly Guid showHelpCmdGuid = new Guid("4A9B7E50-AA16-11D0-A8C5-00A0C921A4D2");
+        private const int showHelpCmdId = 598;
+
+        [ImportingConstructor]
+        public ShowHelpFromErrorList([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            IEducation education, IErrorListHelper errorListHelper)
+            : this(serviceProvider, education, errorListHelper, ThreadHandling.Instance)
+        {
+        }
+
+        internal ShowHelpFromErrorList(IServiceProvider serviceProvider, IEducation education, IErrorListHelper errorListHelper, IThreadHandling threadHandling)
+        {
+            this.education = education;
+            this.errorListHelper = errorListHelper;
+            this.threadHandling = threadHandling;
+
+            InitializeInterception(serviceProvider);
+        }
+
+        internal void InitializeInterception(IServiceProvider serviceProvider)
+        {
+            threadHandling.ThrowIfNotOnUIThread();
+
+            var priority = (IVsRegisterPriorityCommandTarget)serviceProvider.GetService(typeof(SVsRegisterPriorityCommandTarget));
+
+            var command = new CommandID(showHelpCmdGuid, showHelpCmdId);
+
+            var interceptor = new CommandInterceptor(command, () => HandleInterception());
+
+            priority.RegisterPriorityCommandTarget(0, interceptor, out uint cookie);
+        }
+
+        /// <summary>
+        /// If the currently selected error code is a Sonar code show the rule description and
+        /// stop progression to next command else move on.
+        /// </summary>
+        /// <returns></returns>
+        internal CommandProgression HandleInterception()
+        {
```

I was about to add an "If language is null" check but then I realized I wasn't sure about the desired behavior, can I assume that "ShowRuleDescription" will handle it? And if not do we want to throw an exception or just return?